torch/csrc/jit/register_special_ops.cpp

```diff
@@ -127,240 +150,265 @@ void recursiveStore(char* data, const std::vector<int64_t>& sizes, const c10::Ar
+               AT_WARN("Format options are not supported.");
+             }
+
+             auto args = last(stack, num_inputs - 1);
+             std::stringstream ss;
+             for (size_t begin = 0, used_args = 0; true; ++used_args) {
+               size_t loc = format.find("{}", begin);
+               if (loc == std::string::npos) {
+                 ss << format.substr(begin);
+                 break;
+               }
+               ss << format.substr(begin, loc - begin);
+               if (used_args >= args.size()) {
+                 AT_ERROR("Too few arguments for format string: ", format);
+               }
+               ss << args[used_args];
+               begin = loc + 2;
+             }
+
+             drop(stack, num_inputs);
+             push(stack, ss.str());
+             return 0;
+           };
+         }),
+
+#define DEFINE_TORCH_TENSOR_OP(operator_type, c_type, tensor_creation_op)     \
+  Operator(                                                                   \
+      "aten::tensor(" #operator_type                                          \
+      " t, *, ScalarType? dtype=None, Device? device=None"                    \
+      ", bool requires_grad=False) -> Tensor",                                \
+      [](const Node* node) {                                                  \
+        auto initial_scalar_type =                                            \
+            scalarTypeFromJitType(node->inputs().at(0)->type());              \
+        return [initial_scalar_type](Stack& stack) {                          \
+          c_type scalar_val;                                                  \
+          IValue dtype;                                                       \
+          IValue device;                                                      \
+          bool requires_grad;                                                 \
+          pop(stack, scalar_val, dtype, device, requires_grad);               \
+          auto tensor = autograd::make_variable(tensor_creation_op);          \
+          at::ScalarType scalar_type =                                        \
+              dtype.isNone() ? tensor.scalar_type() : dtype.toScalarType();   \
+          c10::Device dev =                                                   \
+              device.isNone() ? tensor.device() : device.toDevice();          \
+          if (scalar_type != initial_scalar_type || dev != tensor.device()) { \
+            tensor = tensor.to(dev, scalar_type);                             \
+          }                                                                   \
+          tensor.set_requires_grad(requires_grad);                            \
+          push(stack, tensor);                                                \
+          return 0;                                                           \
+        };                                                                    \
       }),
```

Comment:
I think the formatting would look less crazy here if the macro didn't include the trailing `,` so the uses would look like:
```
DEFINE_TORCH_TENSOR_OP(…), // <-- trailing comma tells clang-format it's part of a list
DEFINE_TORCH_TENSOR_OP(…), 
...
```